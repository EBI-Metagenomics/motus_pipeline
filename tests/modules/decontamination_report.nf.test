nextflow_process {

    name "Test Process DECONTAMINATION_REPORT"
    script "modules/decontamination.nf"
    process "DECONTAMINATION_REPORT"

    test("Decontamination PE reads count") {

        when {
            params {
                reads = "${baseDir}/tests/modules/fixtures/bwa/mgtest_{1,2}.fastq.gz"
                mode = "paired"
            }
            process {
                """
                input[0] = params.mode
                input[1] = channel.fromFilePairs("${params.reads}", checkIfExists: true).map { it[1] }
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }
    
    test("Decontamination SE reads count") {

        when {
            params {
                reads = "${baseDir}/tests/modules/fixtures/bwa/mgtest_merged.fastq.gz"
                mode = "single"
            }
            process {
                """
                input[0] = params.mode
                input[1] = channel.fromPath("${params.reads}", checkIfExists: true)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }
}