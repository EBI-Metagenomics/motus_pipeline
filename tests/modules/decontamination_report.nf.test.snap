{
    "Decontamination PE reads count": {
        "content": [
            {
                "decontamination_report": [
                    "decontamination_output_report.txt:md5,14864edb3380cc19ac1b2f3abbe13c46"
                ]
            }
        ],
        "timestamp": "2023-03-30T13:22:36+0000"
    },
    "Decontamination SE reads count": {
        "content": [
            {
                "decontamination_report": [
                    "decontamination_output_report.txt:md5,cdd52d64d4c86b2807dfd31c5ee0554a"
                ]
            }
        ],
        "timestamp": "2023-03-30T13:22:36+0000"
    }
}